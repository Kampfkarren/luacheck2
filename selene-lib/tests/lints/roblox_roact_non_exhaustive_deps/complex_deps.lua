local React

local function Component1()
    local a = {}
    React.useEffect(function()
        print(a)
        print(a.b)
        print(a[b])
    end, { a })
end

local function Component2()
    local a = {}
    React.useEffect(function()
        print(a) -- Bad
        print(a.b)
        print(a.b.c)
        print(a.b[c])
    end, { a.b })
end

local function Component()
    local a = {}
    React.useEffect(function()
        print(a) -- Bad
        print(a.b) -- Bad, but already reported with `a`
        print(a.b.c)
        print(a.b.c.d)
        print(a.b.c[d])
    end, { a.b.c })
end

local function Component()
    local a = {}
    React.useEffect(function()
        print(a.b) -- Bad
        print(a.b.c) -- Bad, but already reported with `a.b`
        print(a.b.c.d)
    end, { a.b.c.d })
end

local function Component()
    local a = {}
    local d = {}
    React.useEffect(function()
        print(a.b.c()) -- Good
        print(a.b.d()) -- Bad
        print(d.e.f.g()) -- Good
    end, { a.b.c, d.e.f })
end

local function Component()
    local a = {}
    React.useEffect(function()
        print(a.b["c d"]["e"])
    end, {})
end

local function Component()
    local a = {}
    React.useEffect(function()
        print(a["b c"]["d"])
    end, { a["b c"]["d"] })
end

local function Component()
    local a = {}
    React.useEffect(function()
        local _ = a["b c"]()
    end, {})
end
