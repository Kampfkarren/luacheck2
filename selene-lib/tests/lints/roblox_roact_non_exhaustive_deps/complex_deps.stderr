error[roblox_roact_non_exhaustive_deps]: react hook useEffect has missing dependency: 'a'
   ┌─ complex_deps.lua:19:10
   │
19 │     end, { a.b })
   │          ^^^^^^^
   │
   = help: either include it or remove the dependency array

error[roblox_roact_non_exhaustive_deps]: react hook useEffect has missing dependency: 'a'
   ┌─ complex_deps.lua:30:10
   │
30 │     end, { a.b.c })
   │          ^^^^^^^^^
   │
   = help: either include it or remove the dependency array

error[roblox_roact_non_exhaustive_deps]: react hook useEffect has missing dependency: 'a.b'
   ┌─ complex_deps.lua:39:10
   │
39 │     end, { a.b.c.d })
   │          ^^^^^^^^^^^
   │
   = help: either include it or remove the dependency array

error[roblox_roact_non_exhaustive_deps]: react hook useEffect has missing dependency: 'a.b'
   ┌─ complex_deps.lua:49:10
   │
49 │     end, { a.b.c, d.e.f })
   │          ^^^^^^^^^^^^^^^^
   │
   = help: either include it or remove the dependency array

